### PR TITLE
helm: support for setting internalTrafficPolicy in k8s services

### PR DESCRIPTION
**Pull request by kovaxur**
#### What this PR does

Allows users to add internalTrafficPolicy to the Kubernetes Service of Ingester and Distributor.

#### Checklist

- [N/A] Tests updated.
- [X] Documentation added.
- [X] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [N/A] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Comment by narqo** [comment]
Wouldn't it be equally valuable to have this feature in other mimir services, e.g. in queriers or gateway?

**Comment by kovaxur** [comment]
> Wouldn't it be equally valuable to have this feature in other mimir services, e.g. in `querier`, `gateway`, etc?

Yeah, could be useful, I can add it to the rest. I think the most important is for the ingress direction.

File: operations/helm/charts/mimir-distributed/templates/memcached/_memcached-svc.tpl [hunk 1]
```diff
@@ -17,7 +17,10 @@ metadata:
     {{- toYaml .service.annotations | nindent 4 }}
   namespace: {{ $.ctx.Release.Namespace | quote }}
 spec:
-  type: ClusterIP
+  type: {{ .service.type }}
+  {{- if semverCompare ">= 1.22-0" (include "mimir.kubeVersion" .) }}
+  internalTrafficPolicy: {{ .service.internalTrafficPolicy }}
+  {{- end }}
   clusterIP: None
```
Comment (narqo) [inline operations/helm/charts/mimir-distributed/templates/memcached/_memcached-svc.tpl:24]: I wonder if we need to update services that are explicitly `headless` (note the `ClusterIP: None`).

Comment (kovaxur) [reply operations/helm/charts/mimir-distributed/templates/memcached/_memcached-svc.tpl:24]: True, it doesn't make sense there.

https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/2086-service-internal-traffic-policy/README.md
`The internalTrafficPolicy field will not apply for headless Services or Services of type ExternalName.`

**Comment by narqo** [comment]
Overall, the changes look good to me. But the CI's found that this broke some known use-cases. Try running [`ct`][1] (from the mimir project's root):

```
% ct lint --config operations/helm/ct.yaml --check-version-increment=false
```

Alternatively, this also seems to show the problem:

```
# From the operations/helm/charts/mimir-distributed directory
% helm template . -f ./ci/test-enterprise-configmap-values.yaml
```

[1]: https://github.com/helm/chart-testing

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 ### Grafana Mimir
 
+* [FEATURE] Allow users to set type and internalTrafficPolicy setting on Kubernetes services
```
Comment (narqo) [inline CHANGELOG.md:7]: Oh, I didn't notice this initially. The Helm chart has its own changelog in `operations/helm/charts/mimir-distributed/CHANGELOG.md`. I will fix it

**Review by narqo — APPROVED** [review]
🔥 Works for me. Thank you.

**Pull request merged**